mobile/src/main/java/org/openhab/habdroid/core/OpenHABTracker.java

```diff
@@ -203,10 +203,7 @@ protected Boolean doInBackground(String... strings) {
                         Log.d(TAG, "Socket connected");
                         s.close();
                         return true;
-                    } catch (MalformedURLException e) {
-                        Log.e(TAG, e.getMessage());
-                        return false;
-                    } catch (IOException e) {
+                    } catch (Exception e) {
```

Comment:
My first attempt was to explicit catch IllegalArgumentException, which was raised, but I decided to catch all exceptions. Two reasons, no different logic is in place for the different exceptions. Secondly, more important, is that if another exception is raised and not handled, the background thread dies. Unhnandled exceptions in threads will bring the app down.